src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -191,10 +194,6 @@ private void OnHeadChange(object? sender, BlockReplacementEventArgs e)
         {
             try
             {
-                // Clear snapshot
-                _transactionSnapshot = null;
-                _blobTransactionSnapshot = null;
```

Comment:
Different question, can we make those two use fixed memory?
Basically we should be able to allocate something ~max tx pool count and only update it when needed, instead of allocating it all the time.

Reply:
Don't think so easily; txPool is quite a complicated set of data structures and don't know when the snapshot has finished being used (to change it)

However could use it in one more place

Reply:
Maybe at least pool them?

Reply:
They don't have a very good lifecycle as are multi-use

Would need to introduce a ref counted ArrayPoolList type thing.

Out of scope for this PR though?

Reply:
eh ok :(

Reply:
Something to look at with a bigger refactor of txpool

Reply:
Something to look at with a bigger refactor of txpool vs removing time impact on main processing loop